_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,467 @@
+
+Log in with `Alice` and the password that you set when configuring Keycloak.
+
+Next, you’ll be redirected back to our application, and you should see the "Secured Servlet" page. That means that we
+were able to successfully log in to our application using the Keycloak OpenID provider!
+You will also see the claims that were retrieved using the additional scopes. They were sent through the request object. 
+
+=== Sending the JWT by Reference 
+
+Now try changing the value for *AUTH_REQUEST_FORMAT* to *request_uri* inside the helm chart and keep everything else the same. You can update the openshift deployment using the following commands: 
+```
+helm upgrade oidc-app -f /PATH/TO/ELYTRON/EXAMPLES/elytron-oidc-request/charts/helm.yaml wildfly/wildfly
+```
+
+If the builds don't start automatically, you might have to start them manually on the openshift console. Wait for the build to finish and access the application URL again in a new window. You will see the *request_uri* field appear in the
+url. The `request_uri` parameter is used to send the JWT by reference. The Elytron client sends a PAR request to the
+Pushed Authorization Request Endpoint ({provider-url}/protocol/openid-connect/ext/par/request),
+which creates the request_uri given the JWT Request Object. Once the reference has been made it is only valid for a
+certain amount of time specified in the structure returned by the PAR request. After which the request_uri needs to be
+regenerated. To learn more about the specifications of the Request Object, read the
+https://openid.net/specs/openid-connect-core-1_0.html#RequestUriParameter[OpenID documentation] on passing a Request
+Object by reference.
+
+=== Note about Keystores
+
+You can follow the same instructions to configure your server to use a PKCS12 type keystore. For Keycloak, the signing
+algorithms available are "PS384", "ES384", "RS384", "HS256", "HS512", "ES256", "RS256", "HS384", "ES512","PS256",
+"PS512", "RS512" and "none". If you use algorithms that start with "RS" and "PS" to sign the JWT, you will need to use
+an RSA key pair. For "ES" type keys, use Elliptical curve keys and as mentioned above, adjust the group name for the
+PKCS12 keystore keys to match the SHA value of the algorithms. "none" does not require a keystore and lastly, "HS" keys
+require a symmetric key, where the same secret hash is used by the client and the server to sign and verify respectively. 
+Not all algorithms are supported by all OpenID Providers. Review the documentation and/or the metadata for your OpenID provider to learn more about the supported algorithms. 
+
+== Summary
+
+This example has demonstrated how to secure a web application deployed to WildFly by sending the request parameters as a
+JWT. For more details on the `elytron-oidc-client` subsystem, please check out the
+https://docs.wildfly.org/33/Admin_Guide.html#Elytron_OIDC_Client[documentation] and for more details on OpenID Connect,
```

Comment:
Is there a link somewhere to the new section that you've added to the docs?

Reply:
I have updated the link. 
Thanks @fjuma!